packages/rest/src/coercion/coerce-parameter.ts

```diff
@@ -62,18 +62,13 @@ export function coerceParameter(
       break;
     case 'boolean':
       coercedResult = isTrue(data) ? true : isFalse(data) ? false : undefined;
+      break;
     case 'string':
     case 'password':
     // serialize will be supported in next PR
     case 'serialize':
-      break;
-    case 'unknownType':
     default:
-      throw new HttpErrors.NotImplemented(
-        `Type ${schema.type} with format ${
-          schema.format
-        } is not a valid OpenAPI schema`,
-      );
+      break;
   }
   return coercedResult;
```

Comment:
I think you are missing call of `validator.validateParamAfterCoercion(data)`?

If that's not needed, then I would prefer to see all `break` statements replaced with `return coercedResult`. The next refactoring is to get rid of `let coercedResult` entirely.

Reply:
@bajtos I only applied the `isNaN(coercedResult)` to type number in this PR, that's why it's only called for `case 'number'`